xtask/codegen/css.ungram

```diff
@@ -62,17 +63,43 @@ CssRule =
 
 // .header, .app {}
 // ^^^^^^^^^^^^^
-CssSelectorList = (AnyCssSelectorPattern (',' AnyCssSelectorPattern)*)
-
-AnyCssSelectorPattern =
-	CssIdSelectorPattern
-	| CssClassSelectorPattern
-	| CssUniversalSelectorPattern
-	| CssAttributeSelectorPattern
-	| CssPseudoClassSelectorPattern
-	| CssTypeSelectorPattern
-	| CssCombinatorSelectorPattern
-	| CssBogusPattern
+CssSelectorList = (AnyCssSelector (',' AnyCssSelector)*)
+
+AnyCssSelector =
+	CssComplexSelector
+	| CssCompoundSelector
+	| CssBogusSelector
+
+// div a {}
+// ^^^^^
+// div > a {}
+// ^^^^^^^
+CssComplexSelector =
+	left: AnyCssSelector
+	combinator: CssComplexSelectorCombinator
+	right: AnyCssSelector
+
+CssComplexSelectorCombinator =
+	'>' | '+' | '~' | 'css_space_literal'
+
+CssCompoundSelector =
+	nesting_selector: '&'?
```

Comment:
My suggestion is to **wait** to start implementing the CSS Nesting. It seems that the nesting selector can also be applied after a selector:

```css
.card {
  /* .card styles */
  .featured & & & {
    /* .featured .card .card .card styles */
  }
}
```

Source: https://developer.mozilla.org/en-US/docs/Web/CSS/Nesting_selector#appending_the_nesting_selector

Reply:
I thought that it would a complex selector:
https://swc-css.netlify.app/?code=eJzTS04sSlGo5lJQ0NdS0ANziksqc1KLFbT0gYJ6aamJJaVFqSkKamAIUghRCpeAaEImkQ2o5aoFAGf2HCU

https://prettier.io/playground/#N4Igxg9gdgLgprEAuEA6MBDATgEwATAA6UeeqAZnBjAK5Zz4BkBxpAvsWyADQgQAOMAJbQAzslDYsEAO4AFbAnEoMAGxkYAnuN4AjLBjABrODADKGALZwAMkKhxk5NaLh6Dx02f6H7Ac2QYLBo3EFdLIUDg0NF-VTgARRoIeCcXUIArUQAPMzjE5NSkZ1VXXgBHQrg5aX5lEAxRAFoHBgYeECCMIVV-AGEIS0sMZAbVVQ7YqD94gEEYIKFdGng5OCw7BzTS0IALGEtVAHVdoXhRHzA4MyUzoQA3M81RsFEdEHuQgEkoHARzMBYISCWa-MwwTTxbZlED8aSuI4GfijOFwVxYe6OXj2dEwGoYPzDaGhHxYdEvN4dOH2GBHIQ4GC7ZAADgADLx6JUhPR8YSRsV0rwYBhdHSGUykAAmXg0VwAFRFyhKMLgll0DD+OBsGGmNAJcAAYhAsMMFv5RhgVhAQGw2EA

What do you think?

Reply:
Do we have a link to the official spec?